### PR TITLE
feat: add `@nuxtjs/i18n` package

### PR DESCRIPTION
**Pull request by boojack**
image

### CODE REVIEW

**Comment by tianzhou** [comment]
I am thinking to have a default path which is english and only have chinese if there is override.

So we don't need to duplicate the doc when we only have english doc for now, otherwise the update is too painful.

**Review by d-bytebase — APPROVED** [review]

**Comment by boojack** [comment]
> I am thinking to have a default path which is english and only have chinese if there is override.
> 
> So we don't need to duplicate the doc when we only have english doc for now, otherwise the update is too painful.

We have lots of translating works to do for i18n. So this PR is just trying to add the `@nuxtjs/i18n` package to the project and shows a example. And there is no obviously entry for `zh-CN` version website.

**Review by tianzhou — APPROVED** [review]
LGTM

**Comment by tianzhou** [comment]
> > I am thinking to have a default path which is english and only have chinese if there is override.
> > So we don't need to duplicate the doc when we only have english doc for now, otherwise the update is too painful.
> 
> We have lots of translating works to do for i18n. So this PR is just trying to add the `@nuxtjs/i18n` package to the project and shows a example. And there is no obviously entry for `zh-CN` version website.

Just re-iterate the overall plan

* Only create a Chinese front page
* Keep the existing path for the english version. The chinese version at /zh, 404 if user inputs /zh/xxx while xxx does not have chinese version.

Optional (depending on our bandwidth)
* Translate blog page, we can show the Chinese article in the chinese version.
* A special chinese landing page (e.g. include wechat QR code there)

File: components/DocsSidebar.vue [hunk 1]
```diff
@@ -98,6 +98,11 @@ import { headerDocumentSuffix } from "~/common/const";
 import { useStore } from "~/store";
 import { ContentDocument, Document, DocumentTreeNode } from "~/types/docs";
 
+// `/zh/path/to/file` -> `/path/to/file`
```
[the remark this answers is not in the document]
Comment (boojack) [reply components/DocsSidebar.vue]: This function is not about mapping thing. The raw doc path contains a locale prefix which cannot be used in `localePath` function. So I have to move it firstly.

And I think there is a default mapping for English to Chinese websites: `/*` -> `/zh/*`

**Comment by boojack** [comment]
> Only create a Chinese front page

Maybe we should create a i18n front page with `locales/*.yml` not just only Chinese?

> Keep the existing path for the english version. The chinese version at /zh, 404 if user inputs /zh/xxx while xxx does not have chinese version.

This's the logic right now.

**Comment by tianzhou** [comment]
> @tianzhou
> 
> > Only create a Chinese front page
> 
> Maybe we should create a i18n front page with `locales/*.yml` not just only Chinese?
> 

Oh, you are right.

**Comment by boojack** [comment]
PTAL

**Review by tianzhou — APPROVED** [review]
LGTM

**Pull request merged**